### PR TITLE
[SC] Add test for an empty module

### LINKED ISSUE #2357: [SC] Test deploying a contract module with no types

### PR DESCRIPTION
**Pull request by rowandh**
Closes #2357.

### CODE REVIEW

File: src/Stratis.SmartContracts.IntegrationTests/ContractExecutionFailureTests.cs [hunk 1]
```diff
@@ -469,5 +471,61 @@ public void ContractTransaction_Create_IncorrectParameters()
...
+        {
+            // Ensure fixture is funded.
+            this.node1.MineBlocks(1);
+
+            double amount = 25;
+            Money senderBalanceBefore = this.node1.WalletSpendableBalance;
+            uint256 currentHash = this.node1.GetLastBlock().GetHash();
+
+            // Create transaction with empty module
+            var module = ModuleDefinition.CreateModule("SmartContract", ModuleKind.Dll);
```
Comment (codingupastorm) [inline src/Stratis.SmartContracts.IntegrationTests/ContractExecutionFailureTests.cs]: Would be nice to see an `Assert(module.Types.Count == 1); Assert(module.Types[0].Name == "<Module>")` just to clarify what CreateModule is doing.

Comment (rowandh) [reply src/Stratis.SmartContracts.IntegrationTests/ContractExecutionFailureTests.cs]: Good idea, added
